x/staking/types/authz.go

```diff
@@ -68,13 +80,16 @@ func (authorization StakeAuthorization) Accept(msg sdk.ServiceMsg, block tmproto
 	isValidatorExists := false
 	allowedList := authorization.GetAllowList().GetAddress()
 	for _, validator := range allowedList {
+		ctx.GasMeter().ConsumeGas(gasCostPerIteration, "stake authorization")
```

Comment:
how do external clients know how much this number is? Are they required to dive into the code?

Reply:
They are not able to know exact number of gas. It would require a simulated run. 
You can't do it, because one message can trigger many operations. And I don't think it would make sense to give exact numbers to clients. However, we should provide some estimates based on simulations.